File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = []
    for length in range(1, len(elements) + 1):
        for i in range(len(elements)):
            end = i+length
            if end <= len(elements):
                answer.append(sum(elements[i:end]))
            else:
                answer.append(sum(elements[i:]+elements[:end-len(elements)]))
    return len(set(answer))

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const len = elements.length;
    const answer = []
    
    for(let i = 1; i <= len; i++) {
        for(let n = 0; n < len; n++) {
            const slice = elements.slice(n, n+i)
            const sliceLength = slice.length
            
            if(sliceLength < i) {
                slice.push(...elements.slice(0, i - sliceLength))
            }
            answer.push(slice.reduce((a,b) => a + b))
        }
    }
    
    return new Set(answer).size
}